dowsing/setuptools/setup_py_parsing.py

```diff
@@ -178,7 +178,9 @@ def visit_Call(self, node: cst.Call) -> Optional[bool]:
     BOOL_NAMES = {"True": True, "False": False, "None": None}
     PRETEND_ARGV = ["setup.py", "bdist_wheel"]
 
-    def evaluate_in_scope(self, item: cst.CSTNode, scope: Any) -> Any:
+    def evaluate_in_scope(
+        self, item: cst.CSTNode, scope: Any, target_name: str = ""
```

Comment:
Consider making `target_name` a `List[str]`; imagine longer cycles like (omitting the extra assignments to make this valid and not a 100% repro)

```
import setuptools
c = a
b = a
a = b
setup(name=a)
```

I can't come up with a plausible example where scope needs to be in the list, but maybe...

Reply:
I think this is better handled by tracking target line numbers, and only considering assignments that happen before that target line. This ends up removing the need to track names, and prevents any chance of circular reference or infinite recursion.